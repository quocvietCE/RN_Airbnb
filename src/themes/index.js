import colors from './colors';
import Fonts from './fonts';
// import Metrics from './metrics';
import stylesApp from './stylesApp';

// export {colors, Fonts, Metrics, stylesApp};

export {colors, stylesApp, Fonts};
