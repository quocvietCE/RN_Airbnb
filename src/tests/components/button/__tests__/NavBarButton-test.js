import 'react-native';
import React from 'react';
import NavBarButton from '../../../../components/buttons/NavBarButton';
import Icon from 'react-native-vector-icons/FontAwesome';
import {colors} from '../../../../themes';

import TestRenderer from 'react-test-renderer';
// import {shallow, configure} from 'enzyme';
// import toJson from 'enzyme-to-json';
// import Adapter from 'enzyme-adapter-react-16';

// configure({adapter: new Adapter()});

// const navigation = {
//   navigate: jest.fn(),
// };

describe('Test NavBarButton', () => {
  const mockOnPress = jest.fn();
  const componentTreeIcon = TestRenderer.create(
    <NavBarButton
      location="left"
      color={colors.white}
      icon={<Icon name="angle-left" color={colors.white} size={30} />}
      handleButtonPress={mockOnPress}
    />,
  );

  const componentTreeText = TestRenderer.create(
    <NavBarButton
      location="right"
      text="Forgot Password"
      handleButtonPress={mockOnPress}
    />,
  );

  describe('Rendering NavBarButton', () => {
    it('should match to snapshot with Icon', () => {
      const componentSnapshotTreeIcon = componentTreeIcon.toJSON();
      expect(componentSnapshotTreeIcon).toMatchSnapshot();
    });
    it('should match to snapshot with Text', () => {
      const componentSnapshotTreeText = componentTreeText.toJSON();
      expect(componentSnapshotTreeText).toMatchSnapshot();
    });
  });

  //   describe('Rendering NavBarButton with Enzyme', () => {
  //     const wrapper = shallow(
  //       <NavBarButton
  //         location="left"
  //         color={colors.white}
  //         icon={<Icon name="angle-left" color={colors.white} size={30} />}
  //         handleButtonPress={mockOnPress}
  //       />,
  //     );

  //     it('should match to snapshot with Text', () => {
  //       expect(wrapper).toMatchSnapshot();
  //       console.log('wrapper: ', wrapper);

  //       wrapper.setProps({
  //         icon: <Icon name="angle-left" color={colors.white} size={30} />,
  //       });

  //       console.log('wrapper1: ', wrapper);

  //       expect(wrapper).toMatchSnapshot();

  //       const component = wrapper.dive();
  //       console.log('component2: ', component);
  //       expect(toJson(component)).toMatchSnapshot();
  //       console.log('toJson(component): ', toJson(component));
  //     });
  //   });
});
