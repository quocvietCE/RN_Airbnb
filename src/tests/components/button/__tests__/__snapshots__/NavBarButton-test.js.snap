// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NavBarButton Rendering NavBarButton should match to snapshot with Icon 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="navBarButton"
>
  <View
    style={
      Object {
        "marginLeft": 20,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï„„
    </Text>
  </View>
</View>
`;

exports[`Test NavBarButton Rendering NavBarButton should match to snapshot with Text 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="navBarButton"
>
  <Text
    style={
      Array [
        Object {
          "color": undefined,
        },
        Object {
          "marginRight": 20,
        },
        Object {
          "fontSize": 16,
        },
      ]
    }
  >
    Forgot Password
  </Text>
</View>
`;
